docs/src/content/docs/reference/plugins.md

```diff
@@ -41,12 +47,100 @@ A plugin must provide a unique name that describes it. The name is used when [lo
 
 ## `hooks`
 
-Hooks are functions which Starlight calls to run plugin code at specific times. Currently, Starlight supports a single `setup` hook.
+Hooks are functions which Starlight calls to run plugin code at specific times.
+
+To get the type of a hook's arguments, use the `HookParameters` utility type and pass in the hook name.
+In the following example, the `configSetup` function `options` parameter is typed to match the parameters of the `config:setup` hook:
```

Comment:
Suggesting a small simplification here to help make the sentence a little bit clearer:

```suggestion
In the following example, the `options` parameter is typed to match the arguments passed to the `config:setup` hook:
```